src/primitives.ts

```diff
@@ -927,4 +927,120 @@ export class Wire {
     }
     context.stroke();
   }
-}
\ No newline at end of file
+}
+
+export type DataSample = Map<string, any>
+export class PieChart {
+  pieParts: Array<PiePart>=[];
+
+  constructor(public slicingVariable: string,
+              public dataSamples: DataSample[],
+              public name: string
+              )
+              {
+                this.pieParts = this.definePieParts();
+              }
+
+  public static deserialize(serialized): PieChart { // TODO : clarify this
+    return new PieChart(serialized['slicing_variable'],
+                        serialized['data_samples'],
+                        serialized['name']);
+  }
+
+  private get normedRatio(): number {
+    let sumSamples: number = 0;
+    let normedRatio: number = 2*Math.PI;
+
+    this.dataSamples.forEach(sample => {
+      sumSamples += sample[this.slicingVariable]
+    });
+    normedRatio /= sumSamples;
+    return normedRatio
+  }
+
+  definePieParts(): PiePart[] {
+    const colorRatio: number = 360 / this.dataSamples.length;
+    let initAngle: number = Math.PI/2;
+    let nextAngle: number = initAngle;
+    let colorRadius: number = 0;
+    let pieParts: PiePart[] = [];
+
+    this.dataSamples.forEach(sample => { 
+      colorRadius += colorRatio;
+      nextAngle -= sample[this.slicingVariable] * this.normedRatio;
+
+      let newPart = new PiePart(initAngle, nextAngle, 'hsl('+ colorRadius +', 50%, 50%, 90%)');
+      pieParts.push(newPart);
+
+      initAngle = nextAngle;
+    })
+    return pieParts
+  }
+}
+
+export class PiePart {
+  hidden_color: string = '';
+  path: Path2D;
+  clicked: boolean = false;
+  selected: boolean = false;
+  points_inside: PiePart[] = [this];
```

Comment:
It is really strange to have this kind of things. 
This is a first way to handle the mouse. A better implementation, via a MouseHandler is the purpose of a future dev (https://github.com/Dessia-tech/plot_data/issues/172).

I suggest that we keep the code in commit (174b386b4c308fa2afa4214b4c8d50ef8139a9c2) as a first implementation